music_assistant/providers/listenbrainz_scrobble/__init__.py

```diff
@@ -0,0 +1,173 @@
+            # so they won't be scrobbled
+
+            # https://pylistenbrainz.readthedocs.io/en/latest/api_ref.html#class-listen
+            return Listen(
+                track_name=report.name,
+                artist_name=report.artist,
+                release_name=report.album,
+                recording_mbid=report.mbid,
+            )
+
+        def update_now_playing() -> None:
+            try:
+                listen = make_listen(report)
+                self._client.submit_playing_now(listen)
+                self.logger.debug(f"track {report.uri} marked as 'now playing'")
+                self._currently_playing = report.uri
+            except Exception as err:
+                self.logger.exception(err)
+
+        def scrobble() -> None:
+            try:
+                listen = make_listen(report)
+                listen.listened_at = int(time.time())
+                self._client.submit_single_listen(listen)
+            except Exception as err:
+                self.logger.exception(err)
+
+        # update now playing if needed
+        if self._currently_playing is None or self._currently_playing != report.uri:
+            await asyncio.to_thread(update_now_playing)
+
+        if self.should_scrobble(report):
+            await asyncio.to_thread(scrobble)
+
+        if report.fully_played:
+            # reset currently playing to avoid it expiring when looping songs
+            self._currently_playing = None
+
+    def should_scrobble(self, report: MediaItemPlaybackProgressReport) -> bool:
+        """Determine if a track should be scrobbled, to be extended later."""
+        # ideally we want more precise control
+        # but because the event is triggered every 30s
+        # and we don't have full queue details to determine
+        # the exact context in which the event was fired
+        # we can only rely on fully_played for now
+        return bool(report.fully_played)
+
+
+# configuration keys
+CONF_AUTH_TOKEN = "_auth_token"
+
+
+async def get_config_entries(
+    mass: MusicAssistant,  # noqa: ARG001
+    instance_id: str | None = None,  # noqa: ARG001
+    action: str | None = None,  # noqa: ARG001
+    values: dict[str, ConfigValueType] | None = None,
+) -> tuple[ConfigEntry, ...]:
+    """
+    Return Config entries to setup this provider.
+
+    instance_id: id of an existing provider instance (None if new instance setup).
+    action: [optional] action key called from config entries UI.
+    values: the (intermediate) raw values for config entries sent with the action.
+    """
+    # collect all config entries to show
+    entries: list[ConfigEntry] = [
+        ConfigEntry(
+            key=CONF_AUTH_TOKEN,
+            type=ConfigEntryType.SECURE_STRING,
+            label="Auth Token",
+            required=True,
+            value=values.get(CONF_AUTH_TOKEN) if values else None,
+        ),
+    ]
+
+    return tuple(entries)
```

Comment:
why the intermediate list ?
you can simply directly return the tuple

Reply:
Another left over from the last.fm case, which is more complicated. I'll simplify.